xmovie/core.py

```diff
@@ -358,12 +370,66 @@ def save_frames(self, odir, progress=False):
         elif ~tqdm_avail and progress:
             warnings.warn("Cant show progess bar at this point. Install tqdm")
 
-        for fi in frame_range:
-            fig, ax, pp = self.render_frame(fi)
-            frame_save(
-                fig, fi, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
+        for timestep in frame_range:
+            fig, ax, pp = self.render_single_frame(timestep)
+            save_single_frame(
+                fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
             )
 
+
+
+    def save_frames_parallel(self, odir, progress=False, 
+                             parallel_compute_kwargs=dict()):
+        '''
+        Saves all frames in parallel using dask.map_blocks.
+
+        Parameters
+        ----------
+        odir : path
+            path to output directory
+        progress : Bool
+            Whether or not to show a progress bar.
+        parallel_compute_kwargs : dict
+            Keyword arguments to pass t dask's `compute()` function.
+
+        '''
+        import numpy as np
+        import dask.array as darray
+
+        da = self.data
+        framedim = self.framedim
+        
+        da = da.chunk({self.framedim : 1}) # DataArray needs to have length-1 chunks in framedim
+        chunk_dims = [ dim for dim in da.dims if dim != framedim ] # Create dimensions for each chunk
+               
+
+        total_time = da[framedim]
+        def create_render_save_single_frame2(xr_array, framedim):
+            time_of_chunk = xr_array[framedim]
+            timestep = abs(total_time - time_of_chunk[0]).argmin().item() # get index of chunk in framedim
+
+            fig, ax, pp = self.render_single_frame(timestep)
+            save_single_frame(fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi)
+
+            return time_of_chunk
+
+        mapped_save_and_close_frames = da.map_blocks(func=create_render_save_single_frame2,
+                                                     args=(framedim,),
+                                                     template=xr.ones_like(da[framedim]).chunk({framedim:1}),
+                                                     )
+        if progress:
+            from dask.diagnostics import ProgressBar
```

Comment:
Actually, I might be in favor of removing this completely. This will not show anything for distributed schedulers. The user can always wrap the high level `movie.save(..)` call into a dask ProgressBar. We should just say that the`progress` option only applies to `parallel=False`.

Reply:
Yes, I was kinda unsure about that one. It's a shame that dask doesn't provide a unified progress bar for all schedulers though. Although I should say that using a distributed scheduler proved slower to me in my test than using processes.

Reply:
Done!